code/modules/reagents/reagent_containers/pill.dm

```diff
@@ -228,3 +233,51 @@
 	New()
 		..()
 		reagents.add_reagent("hyperzine", 10)
+
+/obj/item/weapon/storage/pill_bottle/time_release
+	name = "controlled release pill bottle"
+	desc = "A bottle containing special pills which can be calibrated for delayed release with sugar."
+
+/obj/item/weapon/storage/pill_bottle/time_release/New()
+	..()
+	for(var/i=1 to 7)
+		new /obj/item/weapon/reagent_containers/pill/time_release(src)
+
+/obj/item/weapon/reagent_containers/pill/time_release
+	name = "time release pill"
+	desc = "A pill which will not be metabolized until all of the sugar inside metabolizes. Unlike other pills, it is specially designed to be compatible with droppers and syringes."
+	icon_state = "pill18"
+
+/obj/item/weapon/reagent_containers/pill/time_release/injest(mob/M as mob)
+	if(!reagents) return
+	if(!M) return
+	var/timer = round(reagents.get_reagent_amount("sugar"),1)
+	forceMove(M)
+	spawn(timer*30)

```

Comment:
Just making sure, thats normally why we use some defines like 3 SECONDS or 3 MINUTES